src/net_processing.cpp

```diff
@@ -5336,7 +5380,7 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
             LOCK2(m_mempool.cs, peer->m_block_inv_mutex);
 
             size_t reserve = INVENTORY_BROADCAST_MAX_PER_1MB_BLOCK * MaxBlockSize() / 1000000;
-            if (RelayAddrsWithPeer(*peer)) {
+            if (!pto->IsBlockOnlyConn()) {
```

Comment:
this feels wrong; 21528 doesn't introduce any usage of IsBlockOnlyConn? why not use peer->m_addr_relay_enabled?

Reply:
This has already been explained in the PR description but the switchover to `IsBlockOnlyConn` has been split into a separate commit to avoid association with changes from the backport.